### PR TITLE
SSCS-10638-DeadLetterQueueAlert

### PR DESCRIPTION
**Pull request by rmurphy-hmcts**
### JIRA link (if applicable) ###
SSCS-10638-DeadLetterQueueAlert

### Change description ###
Action group and alert metric added.
Ability to turn on/off metrics in tfvar files

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by reespozzi — APPROVED** [review]

File: demo.tfvars [hunk 1]
```diff
@@ -2,3 +2,45 @@ tribunals_frontend_external_hostname = "sscs-tribunals-frontend-demo.service.cor
...
+        metric_namespace = "Microsoft.ServiceBus/namespaces"
+        metric_name      = "DeadletteredMessages"
+        aggregation      = "Average"
+        operator         = "GreaterThan"
+        threshold        = 0
+        dimension = [
+          {
+            name     = "EntityName"
+            operator = "Include"
+            values   = ["mohammeds-awesome-queue"]
```
Comment (MohammedHMCTS) [inline demo.tfvars:35]: Same here

File: servicebus.tf [hunk 2]
```diff
@@ -47,4 +47,6 @@ module "notifications-subscription" {
   namespace_name      = local.servicebus_namespace_name
   resource_group_name = local.resource_group_name
   topic_name          = local.evidenceshare_topic_name
+
+  depends_on = [module.evidenceshare-topic]
```
Comment (MohammedHMCTS) [inline servicebus.tf:51]: Not to sure if this is required?

**Review by MohammedHMCTS — CHANGES_REQUESTED** [review]

File: servicebus.tf (hunk 2 above)
Comment (rmurphy-hmcts) [reply servicebus.tf:51]: This is required due to the topic needing to be created before the module can run. It has already been incorporated into master branch.

**Review by thomaswatsonv1 — APPROVED** [review]
Looks good!

**Review by MohammedHMCTS — APPROVED** [review]

**Pull request merged**